core/src/main/scala/kafka/raft/RaftManager.scala

```diff
@@ -214,6 +220,12 @@ class KafkaRaftManager[T](
     KafkaRaftManager.createLogDirectory(new File(config.metadataLogDir), logDirName)
   }
 
+  // visible for testing cleanup
+  private[raft] def deleteDataDir(): Unit = {
+    val logDirName = Log.logDirName(topicPartition)
+    KafkaRaftManager.deleteLogDirectory(new File(config.metadataLogDir), logDirName)
```

Comment:
@jsancio What do you think about exposing this api to the package for cleaning up log directories in tests? I'm not sure exactly how to override `createDataDir` with `tempDir` from the tests. It is called internally in `KafkaRaftManager` without an argument.

Reply:
Would this class need to extend KafkaRaftManager to override the directory field in that class with tempDir?

Reply:
This code should not be in `core/src/main` since it is not used by `core/src/main`.